app/src/main/java/org/mozilla/fenix/quickactionsheet/QuickActionSheetController.kt

```diff
@@ -49,18 +50,22 @@ class DefaultQuickActionSheetController(
 
     override fun handleBookmark() {
         context.metrics.track(Event.QuickActionSheetBookmarkTapped)
-        bookmarkTapped(currentSession)
+        sessionManager.selectedSession?.let {
+            bookmarkTapped(it)
+        }
     }
 
     override fun handleOpenLink() {
         context.metrics.track(Event.QuickActionSheetOpenInAppTapped)
 
         val getRedirect = appLinksUseCases.appLinkRedirect
-        val redirect = currentSession.let {
+        val redirect = sessionManager.selectedSession?.let {
             getRedirect.invoke(it.url)
         }
 
-        redirect.appIntent?.flags = Intent.FLAG_ACTIVITY_NEW_TASK
-        appLinksUseCases.openAppLink.invoke(redirect)
+        redirect?.appIntent?.flags = Intent.FLAG_ACTIVITY_NEW_TASK
```

Comment:
nit: move these inside the `sessionManager.selectedSession?.let {` block so we don't need multiple null checks